proxy/src/main/java/org/apache/rocketmq/proxy/service/sysmessage/HeartbeatSyncer.java

```diff
@@ -0,0 +1,224 @@
+                        consumerGroup, clientChannelInfo, consumeType, messageModel, consumeFromWhere, subList, t);
+                }
+            });
+        } catch (Throwable t) {
+            log.error("heartbeat submit register broadcast failed. group:{}, clientChannelInfo:{}, consumeType:{}, messageModel:{}, consumeFromWhere:{}, subList:{}",
+                consumerGroup, clientChannelInfo, consumeType, messageModel, consumeFromWhere, subList, t);
+        }
+    }
+
+    public void onConsumerUnRegister(String consumerGroup, ClientChannelInfo clientChannelInfo) {
+        if (clientChannelInfo == null || ChannelHelper.isRemote(clientChannelInfo.getChannel())) {
+            return;
+        }
+        try {
+            this.threadPoolExecutor.submit(() -> {
+                try {
+                    ProxyConfig proxyConfig = ConfigurationManager.getProxyConfig();
+                    RemoteChannel remoteChannel = RemoteChannel.create(clientChannelInfo.getChannel());
+                    if (remoteChannel == null) {
+                        return;
+                    }
+                    HeartbeatSyncerData data = new HeartbeatSyncerData(
+                        HeartbeatType.UNREGISTER,
+                        clientChannelInfo.getClientId(),
+                        clientChannelInfo.getLanguage(),
+                        clientChannelInfo.getVersion(),
+                        consumerGroup,
+                        null,
+                        null,
+                        null,
+                        proxyConfig.getLocalServeAddr(),
+                        remoteChannel.encode()
+                    );
+
+                    log.debug("sync unregister heart beat. topic:{}, data:{}", this.getBroadcastTopicName(), data);
+                    this.sendSystemMessage(data);
+                } catch (Throwable t) {
+                    log.error("heartbeat unregister broadcast failed. group:{}, clientChannelInfo:{}, consumeType:{}",
+                        consumerGroup, clientChannelInfo, t);
+                }
+            });
+        } catch (Throwable t) {
+            log.error("heartbeat submit unregister broadcast failed. group:{}, clientChannelInfo:{}, consumeType:{}",
+                consumerGroup, clientChannelInfo, t);
+        }
+    }
+
+    @Override
+    public ConsumeConcurrentlyStatus consumeMessage(List<MessageExt> msgs, ConsumeConcurrentlyContext context) {
+        if (msgs == null || msgs.isEmpty()) {
+            return ConsumeConcurrentlyStatus.CONSUME_SUCCESS;
+        }
+
+        for (MessageExt msg : msgs) {
+            try {
+                HeartbeatSyncerData data = JSON.parseObject(new String(msg.getBody(), StandardCharsets.UTF_8), HeartbeatSyncerData.class);
+                if (data.getConnectProxyIp().equals(ConfigurationManager.getProxyConfig().getLocalServeAddr())) {
+                    continue;
+                }
```

Comment:
If there are two proxy instances on one machine, will there be a problem? The same address cannot distinguish different heartbeat data, maybe we need a unique proxyId?

Reply:
fixed